feat(new-trace): Updating canFetch property of transactions based on spans count

Abdkhan14 opened the pull request:
Note: For testing, try out next.js traces, they usually have the Missing Spans Case.

Customers are confused by the `Missing spans `scenario, for which we currently render a `No Data` node and the following CTA:
![Screenshot 2024-08-12 at 1 14 48 PM](https://github.com/user-attachments/assets/6c6a07e2-f8b7-433e-80ef-7ca7a685ad73)

We've decided to make transactions that don't have embedded spans unable to fetch: We wont render the '+' icons, they will just be leaf nodes. We still need to decide if we want a CTA for this case, as the sdk team doesn't always see Missing spans as a problem. 

The `events-trace-meta` endpoint now gives us an array of 'transaction_id' -> 'span_children_count' called `transaction_child_count_map` that I used to achieve this new functionality. This means that we need to wait for the meta endpoint response for the tree, which is fine since the endpoint is mostly faster than the trace endpoint: [comparing durations](https://sentry.sentry.io/discover/homepage/?dataset=discover&display=top5&environment=prod&field=transaction&field=p99%28transaction.duration%29&id=16351&interval=1d&name=&project=1&query=transaction%3A%2Fapi%2F0%2Forganizations%2F%7Borganization_id_or_slug%7D%2Fevents-trace%2F%7Btrace_id%7D%2F+or+transaction%3A%2Fapi%2F0%2Forganizations%2F%7Borganization_id_or_slug%7D%2Fevents-trace-meta%2F%7Btrace_id%7D%2F&queryDataset=error-events&sort=-transaction&statsPeriod=14d&topEvents=5&yAxis=p99%28transaction.duration%29)

I will put up two stacked PRs:
- Pass the correct timestamp to the `events-trace-meta` endpoint for replay traces. Otherwise we are currently not getting the correct `transaction_child_count_map` for replay traces.
- Remove all the `No data` node code.

Commit:
feat(new-trace): Updating canFetch property of transactions based on spans count.

Commit:
feat(new-trace): Fixing tests

Abdkhan14 committed:
Merge branch 'master' into abdk/new-trace-can-fetch

Commit:
feat(new-trace): Minor iteration.

Commit:
feat(new-trace): Minor iteration.

Commit:
fix(new-trace): Minor iteration.

Abdkhan14 committed:
Merge branch 'master' into abdk/new-trace-can-fetch

JonasBa commented on static/app/views/performance/newTraceDetails/traceApi/useTraceMeta.tsx:
```diff
@@ -158,15 +184,13 @@ export function useTraceMeta(traceSlugs: string[]): TraceMetaQueryResults {
         performance_issues: 0,
         projects: 1,
         transactions: 1,
+        transactiontoSpanChildrenCount: {'1': 0},
```
Is this needed?

JonasBa reviewed (approved):
Looks good. My one concern is just that we should have some link for transactions without spans somewhere where we either link users to manual instrumentation docs or allow them to provide feedback (this could be global and not related to the clicked node). The reason I think it'd be helpful is because we've received feedback from users who dont know what to do in this case and need guidance(taking that away without providing a fallback seems like a bad choice)

Commit:
fix(new-trace): Minor iteration.

JonasBa commented on static/app/views/performance/newTraceDetails/traceModels/traceTree.tsx:
```diff
@@ -562,7 +568,13 @@ export class TraceTree {
 
-      node.canFetch = true;
+      // We check for at least 2 spans because the first span is the transaction itself.
+      if (isTransactionNode(node)) {
+        const spanChildrenCount =
+          metaResults?.data?.transactiontoSpanChildrenCount[node.value.event_id];
+        node.canFetch = spanChildrenCount ? spanChildrenCount >= 2 : false;
+      }
```
If the result of metaResults.data lookup is undefined, then node.canFetch has to return true. It can only return false if the key is present in the lookup and the value is >= 2

Commit:
fix(new-trace): Minor iteration

Abdkhan14 committed:
Merge branch 'master' into abdk/new-trace-can-fetch

Abdkhan14 committed:
Merge branch 'master' into abdk/new-trace-can-fetch

Commit:
feat(new-trace): Adding custom instrumentation link to transaction details with no span children.

Abdkhan14 committed:
feat(new-trace): Passing the correct timestamp to trace meta endpoint. (#76426)

There is a mismatch between the number of events reflected by the
`events-trace` endpoint and the `events-trace-meta` endpoint. The latter
reports less events. This is a bigger issue now since we use the
response from the `events-trace-meta` endpoint to determine the
`canFetch` property of a txn in the trace waterfall (i.e. to determine
if there are embedded span children and if we should render the '+'
icon).

We query for the `traceId`and `min(timestamp)` of the traces associated
with a replay. We weren't passing `timestamp: min(timestamp)` as a param
to the events-trace-meta endpoint from replay trace tab.

This doesn't solve the problem, the mismatch still exists, but this is
the right thing to do and lets us narrow down the issue to the Backend.

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

Abdkhan14 committed:
chore(new-trace): Removing code for no data or empty node in waterfall. (#76522)

Before: 
<img width="1286" alt="Screenshot 2024-08-25 at 4 23 27 PM"
src="https://github.com/user-attachments/assets/a75c01a8-8f0c-4f27-b057-ae8a96cb1f83">

After, note that we are working on a banner that points to platform
specific custom instrumentation docs in the details for such nodes:
<img width="1285" alt="Screenshot 2024-08-25 at 4 26 23 PM"
src="https://github.com/user-attachments/assets/eccf0050-3678-4c4e-a18b-98b9d6660dc0">

Refer to the parent PR for context:
[link](https://github.com/getsentry/sentry/pull/75849)

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

Commit:
feat(new-trace): Resolving conflicts
